src/components/TokenSelectorModal.tsx

```diff
@@ -2,90 +2,83 @@ import React, { Dispatch, SetStateAction, ReactElement, Fragment, useRef } from
-        initialFocus={closeIconRef}
+  }
+
+  return (
+    <Transition appear show={isOpen} as={Fragment}>
+      <Dialog as={'div'} onClose={() => setIsOpen(false)} className={style.modalWrapper} initialFocus={closeIconRef}>
+        <Transition.Child
+          as={Fragment}
+          enter="ease-out duration-500"
+          enterFrom="opacity-0"
+          enterTo="opacity-100"
+          leave="ease-in duration-200"
+          leaveFrom="opacity-100"
+          leaveTo="opacity-0"
         >
-            <Transition.Child
-            as={Fragment}
-            enter="ease-out duration-500"
-            enterFrom="opacity-0"
-            enterTo="opacity-100"
-            leave="ease-in duration-200"
-            leaveFrom="opacity-100"
-            leaveTo="opacity-0"
-            >
-                <div className={style.modalOverlay}>
-                    <Dialog.Panel className={style.modal}>
-                        <div className={style.headingContainer}>
-                            <Dialog.Title className={style.modalTitle}>Select a Token</Dialog.Title>
-                            <XIcon
-                            className={style.closeIcon}
-                            onClick={() => setIsOpen(false)}
-                            ref={closeIconRef}
-                            />
-                        </div>
-                        <hr className={style.headingBreak}></hr>
-                        <div className={style.tokensContainer}>
-                            {Tokens.map((token, index) => {
-                                return (
-                                    <div
-                                    key={index}
-                                    className={style.tokenContainer}
-                                    onClick={() => handleSelectedToken(token)}
-                                    >
-                                        <img src={token.imgPath} className={style.tokenImg} />
-                                        <div className={style.tokenSymbol}>{token.symbol}</div>
-                                        <div className={style.tokenBalance}>0</div>
-                                    </div>
-                                )
-                            })}
-                        </div>
-                    </Dialog.Panel>
-                </div>
-            </Transition.Child>
-        </Dialog>
-        </Transition>
-    )
+          <div className={style.modalOverlay}>
+            <Dialog.Panel className={style.modal}>
+              <div className={style.headingContainer}>
+                <Dialog.Title className={style.modalTitle}>Select a Token</Dialog.Title>
+                <XIcon className={style.closeIcon} onClick={() => setIsOpen(false)} ref={closeIconRef} />
+              </div>
+              <hr className={style.headingBreak}></hr>
+              <div className={style.tokensContainer}>
+                {Tokens.map((token, index) => {
```

Comment:
Did you remove the otherToken from the list if not null?

Reply:
Didn't get this point? What do you mean by this?

Reply:
Sorry. Can you please elaborate this point

Reply:
When the user opens the dropdown, the "otherToken" can be removed from the list so that they can't put duplicate tokens selected.

Reply:
Done